docs/reference-guide/modules/events/pages/index.adoc

```diff
@@ -1,18 +1,345 @@
-A summary of the various sections is given below.
+== Core concepts
+
+=== `EventSink`
+
+An **EventSink** is any component where you can send events into.
+This is a general abstraction that encompasses:
+
+- **Event Bus** - Distributes events to internal handlers and processors (both sending and consuming)
+- **External system facades** - Publishes events to external systems (Kafka, RabbitMQ, etc.)
+- **Event Store** - Persists events and distributes them to processors
+
+The EventSink abstraction allows you to publish events without knowing the specific distribution mechanism.
+
+For details, see xref:event-dispatchers.adoc[Event Dispatchers].
+
+=== `MessageStream`
+
+Event handling is built on **MessageStream**, a non-blocking, async-native abstraction for working with sequences of messages.
+
+MessageStream provides:
+
+- **Non-blocking operations** - Never blocks waiting for messages
+- **Composable transformations** - Rich set of operators (map, filter, concat, etc.)
+- **Backpressure handling** - Callback-based availability signaling
+- **Lazy evaluation** - Transformations computed only when needed
+
+This architecture enables efficient, scalable event processing without blocking threads.
+
+For more on MessageStream, see xref:messaging-concepts:message-stream.adoc[MessageStream].
+
+=== Dynamic Consistency Boundaries (DCB)
+
+**Dynamic Consistency Boundaries** (DCB) allow flexible grouping of events for consistency guarantees.
+
+DCB enables you to:
+
+- **Tag events** with arbitrary key-value pairs
+- **Query events by tags** using EventCriteria
+- **Define consistency boundaries** based on business needs, not technical constraints
+- **Track consistency markers** to prevent write conflicts
+
+DCB enables more flexible event sourcing patterns while maintaining consistency guarantees.
+
+For details, see xref:infrastructure.adoc[Event Infrastructure].
+
+=== `ProcessingContext`
+
+Event handlers receive a **ProcessingContext** that manages the processing lifecycle:
+
+- **Mutable resource management** - Share resources across processing stages
+- **Lifecycle hooks** - Register callbacks for commit, rollback, cleanup
+- **Correlation data** - Automatic propagation through the system
+
+ProcessingContext is created automatically by the framework and available in all event handlers.
+
+For details, see xref:messaging-concepts:processing-context.adoc[ProcessingContext].
+
+=== Message types
+
+Event identity is determined by **MessageType** (qualified name + version), not by Java class.
+
+This enables:
+
+- **Multiple representations** - Same event, different Java classes
+- **Payload conversion** - Automatic conversion at handling time
+- **Reduced upcasting** - Many evolution scenarios handled by conversion instead
+
+For details, see xref:event-handlers.adoc#message-types-and-event-names[Message Types and Event Names].
+
+== Event handling overview
+
+This section covers the complete event handling capabilities in Axon 5:
```

Comment:
We should be specifying things like "Axon 5" anywhere. The fact that a user lands on this page already means they're here for AF5. No need to add sentences for that.